### PR TITLE
adding TestingServer.stop() to hopefully fix zk transient errors

### PR DESCRIPTION
**Pull request by himanshug**
should fix errors like
```
BrokerServerViewTest.testSingleServerAddedRemovedSegment:112->setupZNodeForServer:396 » NodeExists
```
as seen in https://travis-ci.org/druid-io/druid/jobs/97777343

### CODE REVIEW

File: server/src/test/java/io/druid/curator/CuratorTestBase.java [hunk 1]
```diff
@@ -88,6 +90,12 @@ protected void setupZNodeForServer(DruidServer server, ZkPathsConfig zkPathsConf
   protected void tearDownServerAndCurator()
   {
     CloseQuietly.close(curator);
+    try {
+      server.stop();
```
Comment (guobingkun) [inline server/src/test/java/io/druid/curator/CuratorTestBase.java]: This is called inside ```CloseQuietly.close(server)``` I think.

Comment (himanshug) [inline server/src/test/java/io/druid/curator/CuratorTestBase.java]: its not.

Comment (guobingkun) [inline server/src/test/java/io/druid/curator/CuratorTestBase.java]: Isn't it ```CloseQuietly.close(server)``` -> ```server.close()``` -> ```testingZooKeeperServer.close()``` -> ```testingZooKeeperServer.stop()```?
```server.top()``` -> ```testingZooKeeperServer.stop()```

Comment (himanshug) [inline server/src/test/java/io/druid/curator/CuratorTestBase.java]: yes, you are right. not sure why the test still fails though.

**Comment by drcrallen** [comment]
How appropriate is it to make this a `@Rule` instead of having to explicitly call stop?

**Comment by himanshug** [comment]
@drcrallen stop() is called implicitly on close of the server, so I have closed this PR.

**Pull request closed without merging**